packages/flutter/lib/src/material/banner.dart

```diff
@@ -394,3 +430,51 @@ class _MaterialBannerState extends State<MaterialBanner> {
     );
   }
 }
+
+class _BannerDefaultsM2 extends MaterialBannerThemeData {
+  _BannerDefaultsM2(this.context)
+    : _theme = Theme.of(context),
+      super(elevation: 0.0);
+
+  final BuildContext context;
+  final ThemeData _theme;
+
+  @override
+  Color? get backgroundColor => _theme.colorScheme.surface;
+
+  @override
+  Color? get dividerColor => Theme.of(context).colorScheme.surfaceVariant;
```

Comment:
This broke golden tests for flutter gallery, is this the correct value?

Reply:
Yes, based on token
https://github.com/flutter/flutter/blob/bce91637c6dbe19a76b9836265720bd27a3b9e39/dev/tools/gen_defaults/data/banner.json#L22

Reply:
Perhaps the golden test is broken this color is added to the M2 template in this PR

Which uses Theme.dividerColor

https://github.com/flutter/flutter/blob/bce91637c6dbe19a76b9836265720bd27a3b9e39/packages/flutter/lib/src/material/divider.dart#L137

Reply:
I can revert this PR or remove the divider color from the M2 template.

Reply:
Okay, thanks. Since this is not breaking per the breaking change policy, and is effectively a fix so let's hold off unless there's a need to revert due to Google-internal usages.

Reply:
Thanks for the update!

Reply:
@TahaTesser  Hi Taha, We are reverting this PR in #111288 because it caused some scuba breakages to google3. Please create another PR for this, thanks!

Reply:
Thank you @hangyujin! My bad I should have been careful with the M2 template.